### PR TITLE
Pass exceptions on to callers

### PR DESCRIPTION
**Pull request by slacrherbst**
This PR changes the Rogue approach to handling errors. Previously the errors were caught and logged. In this new model the exceptions are passed on to the caller. This allows the caller to become aware of the errors and avoids some errors being masked, sometimes resulting a sequence of commands continuing event after a fatal error occurs.

With this change, exceptions are now passed across the ZMQ client/server link.

The legacy pyqt GUI will simply log an error. The PYDM interface will open an error dialog box. 

I am still looking at the proper way to deal with exceptions called from the epics3 and epics4 interfaces.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -112,12 +112,13 @@ jobs:
       script:
         # Run tests with coverage
         - cd $TRAVIS_BUILD_DIR
-        - python3 -m pytest --cov
+        - python3 -m pytest
+        #- python3 -m pytest --cov
```
Comment (ruck314) [inline .travis.yml:116]: How come this is committed out?

File: .travis.yml [hunk 2]
```diff
@@ -112,12 +112,13 @@ jobs:
...
 
       after_success:
         # Upload coverage report
-        - codecov
-        - coverage report -m
+        #- codecov
```
Comment (ruck314) [inline .travis.yml:120]: How come this is committed out?

File: .travis.yml (hunk 1 above)
Comment (slacrherbst) [reply .travis.yml:116]: It appears to be broken in the master branch version of the build system. There is a version conflict which will require upgrading travis, which is already in place for pre-release. I figured just to temp get by for now.

File: tests/test_server.py [hunk 3]
```diff
@@ -170,10 +172,10 @@ def _myArray(self):
     with DummyTree() as dummyTree:
         dummyTree.saveAddressMap('test.csv')
 
-        pyrogue.waitCntrlC()
+        #pyrogue.waitCntrlC()
```
[the remark this answers is not in the document]
Comment (slacrherbst) [reply tests/test_server.py]: that was a mistake. I corrected it.

**Comment by jesusvasquez333** [comment]
I'm going to test this one in one of the SMuRF system...

**Comment by jesusvasquez333** [comment]
I tested this changes yesterday and I was able to catch exception thrown during the YAML defaults loading.

**Pull request merged**